guillotina/db/transaction.py

```diff
@@ -384,16 +384,61 @@ def tpc_cleanup(self):
         return await self._manager._storage.get_child(self, container._p_oid, key)
 
     @profilable
-    async def get_child(self, container, key):
-        result = await self._get_child(container, key)
+    async def get_child(self, parent, key):
+        result = await self._get_child(parent, key)
         if result is None:
             return None
 
-        obj = reader(result)
-        obj.__parent__ = container
+        return self._fill_object(result, parent)
+
+    def _fill_object(self, item, parent):
+        obj = reader(item)
+        obj.__parent__ = parent
         obj._p_jar = self
         return obj
 
+    async def _get_batch_children(self, parent, keys):
+        for litem in await self._manager._storage.get_children(
+                self, parent._p_oid, keys):
+            if len(litem['state']) < self._cache.max_cache_record_size:
+                await self._cache.set(litem, container=parent, id=litem['id'])
+                self._cache._stored += 1
+            yield self._fill_object(litem, parent)
+
+    async def get_children(self, parent, keys):
+        '''
+        More performant way to get groups of items.
+        - look at cache
+        - batch get from storage
+        - async for iterate items
+        - store retrieved values in storage
+        '''
+        lookup_group = []  # backlog of object that need to be looked up
+        for key in keys:
+            item = await self._cache.get(container=parent, id=key)
+            if item is None:
+                self._cache._misses += 1
+                lookup_group.append(key)
+                if len(lookup_group) > 15:  # limit batch size
```

Comment:
Why this limit, can it be configured ?

Reply:
async generators in postgresql queries are not ideal. They require cursors. Cursors require an active transaction and we often don't have that.

Reply:
for example, getting all the objects in a large container.